updated ai flows section

rsun19 opened the pull request:
<!--- If this is a non-code change, this template is not required; reference any issues or top-level descriptions as needed -->
<!--- All code change PRs should relate to an issue, reference it here; see example below -->
https://issues.redhat.com/browse/RHOAIENG-22035

## Description
<!--- Describe your changes in detail; the what, the why, any findings, etc -->
<!--- Include any screenshots of changed UI; Include any gifs if it was a flow / UX change -->

Changed the ai flows section by changing the title, icon, and added a hint.

<img width="1704" alt="Screenshot 2025-04-07 at 9 48 52 PM" src="https://github.com/user-attachments/assets/d8704b15-03af-4229-ac08-e5fad349a365" />
<img width="648" alt="Screenshot 2025-04-07 at 9 49 13 PM" src="https://github.com/user-attachments/assets/53de11d8-5e2d-4929-a1c4-23611075d3bf" />
<img width="1115" alt="Screenshot 2025-04-08 at 1 40 21 PM" src="https://github.com/user-attachments/assets/020f7f34-c8f5-45f9-b363-f65a07012fa8" />

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

Go to the home page and enable the fine-tuning feature flag

## Test Impact
<!--- What tests have you done to cover the implemented functionality -->
<!--- If tests are not applicable, explain why here -->

Added cypress tests

## Request review criteria:
<!--- This PR will be merged by any repository approver when it meets all the points in the checklist -->
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 
<!--- You can ignore these if you are doing manifest, backend, internal logic, etc changes; aka non-UI / visual changes -->
- [x] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

claudialphonse78 commented on frontend/src/pages/home/aiFlows/useAIFlows.tsx:
```diff
@@ -88,8 +94,38 @@ export const useAIFlows = (): React.ReactNode => {
+                </Content>
+                <Content>
+                  <Link to={modelCustomizationRootPath} data-testid="ai-flow-hint-navigate">
+                    Learn more about LAB-tuning
```
If possible, maybe you can also add a test case for the redirection from the home page to the model customization page.

rsun19 replied:
Haha I also forgot to close the hint in the test

rsun19 commented:
/retest

claudialphonse78 commented on frontend/src/__tests__/cypress/cypress/pages/home/homeAIFlow.ts:
```diff
@@ -15,6 +15,26 @@ export class HomeAIFlow extends Contextual<HTMLElement> {
   }
 
+  findAIFlowHint(): Cypress.Chainable<JQuery<HTMLElement>> {
+    return cy.findByTestId('ai-flow-hint');
```
one small question, shouldn't we scope the test to this section?
I see it scoped it some files and not in some
else rest all LGTM
`this.find().findByTestId('ai-flow-hint')`

claudialphonse78 commented:
lgtm

DaoDaoNoCode commented on frontend/src/pages/home/aiFlows/AIFlowHint.tsx:
```diff
@@ -0,0 +1,83 @@
+
+  return (
+    <Card data-testid="ai-flow-hint" style={{ borderRadius: 16 }}>
+      <Flex
```
Consider using the `Split` layout probably? And you can create a `Flex` to the title, body and image if you want to make it wrappable. In this way, the close button will always be on the right and you don't need to make 2 separate ones here to show on different breakpoints.

DaoDaoNoCode commented on frontend/src/pages/home/aiFlows/AIFlowHint.tsx:
```diff
@@ -0,0 +1,83 @@
+import { TimesIcon } from '@patternfly/react-icons';
+import { useBrowserStorage } from '~/components/browserStorage';
+
+type AIFlowHintProps = {
+  title: React.ReactNode;
+  body: React.ReactNode;
+  isDisplayed: boolean;
+  image: React.ReactNode;
+};
```
Just curious, will this component be reused and have different contents? If not, why not just put these values into the component?

DaoDaoNoCode reviewed (commented):
Left some comments about the styles and layout.

rsun19 commented:
This is the cleanest I can make it so far. 

<img width="1685" alt="Screenshot 2025-04-15 at 3 36 37 PM" src="https://github.com/user-attachments/assets/bceadabc-0f59-4db0-b5cf-6db57cebf8c7" />
<img width="1156" alt="Screenshot 2025-04-15 at 3 37 15 PM" src="https://github.com/user-attachments/assets/494ba1c3-f72d-4ca9-b369-db6902c02478" />


First, I cannot put the image inline with the text until the `2xl` breakpoint without high margins due to the height and width of the image. And the image, when smaller than `2xl` will not grow excessively, because I think that it would make the hint too big for the user.

Second, I cannot `justify center` the image. The reason is that I put a flex property on the flex items, which disallows the text to 'take over' and push the image out. Consequently, this erases the `justify center` property, but I'm not sure how important this actually is.

Thoughts @DaoDaoNoCode?

DaoDaoNoCode commented:
Actually, the output looks good to me, but I would like to see what UX thinks. @kywalker-rh Hello Kyle, could you check the above screenshots on different breakpoints?

kywalker-rh commented:
Thanks for catching the spacing around the illustration @DaoDaoNoCode!

@rsun19 For the breakpoint you can see the work @tobiastal did in the [Figma file](https://www.figma.com/design/wJh8G8wse5UVzaXkDuU1Tu/Summit-2025-Home-page-and-left-nav-updates?node-id=3209-55146&t=PtvE39XAl64kZD7v-1). The text should wrap until the card gets to 914px wide, then the illustration should move below the text. You can resize the window in that file (just command + z when done), it does end up with padding around the illustration as the width gets smaller, so based on your earlier comment I think that should be alright.

Let me know if it would be helpful to have a quick screen share with @tobiastal or myself as you work through it.

rsun19 commented:
@kywalker-rh Do i need edit permissions to resize the window? I currently don't have it

kywalker-rh commented:
@rsun19 I didn't think you would need it, its supposed to be anyone in RH, but I just added you specifically, so hopefully you can now. If that doesn't work let me know and we can do a screen share or I can record a video.

rsun19 commented:
@kywalker-rh The breakpoint where the image goes under the text in the mocks is set as `914px` wide, however, I could only get it to `992px` since that is a standard `lg` breakpoint for patternfly. I'm wondering how important this is, since I'm not sure if this can be done without adding custom styling. Here's the video of my changes:

https://github.com/user-attachments/assets/d4280f16-0a8c-4c6a-820c-941889a341bf

cc @DaoDaoNoCode

kywalker-rh commented:
@rsun19 I think this looks reasonable to me, it changes at the same time the Projects cards switch to 3 across, so it isn't just changing at what seems like a random break point. @tobiastal Any issues with this behavior?

The only other thing I was thinking as I was looking at it, is it possible to have the illustration centered in the white space to the right of the text? That's how it works in the mocks, but here it looks like the left-side of the illustration is fixed to the center of the entire card.

rsun19 commented:
@kywalker-rh updated here:


https://github.com/user-attachments/assets/deaf6a1c-8ef6-4996-aad8-acc4a3109c36

kywalker-rh commented:
Perfect, thanks @rsun19!

tobiastal commented:
That looks great from my perspective!
cc @kywalker-rh

DaoDaoNoCode reviewed (commented):
Overall LGTM, left one small comment

rsun19 commented on frontend/src/pages/home/aiFlows/AIFlowHint.tsx:
```diff
@@ -0,0 +1,98 @@
+  }
+
+  return (
+    <Card data-testid="ai-flow-hint" style={{ overflowY: 'hidden' }}>
```
Good catch! I suppose my changes made this style obsolete.

rsun19 commented:
@DaoDaoNoCode FYI, the close button did not go all the way right at 3000px, so I added the `isFilled` component to the `SplitItem` and added the `flex end` style to fix this

DaoDaoNoCode reviewed (approved):
/lgtm
Looks great, well done!
